src/imitation/scripts/analyze.py

```diff
@@ -13,13 +17,43 @@
 from imitation.util.sacred import dict_get_nested as get
 
 
+@analysis_ex.capture
+def _gather_sacred_dicts(
+    source_dirs: List[str], run_name: str, env_name: str, skip_failed_runs: bool
+) -> List[sacred_util.SacredDicts]:
+    # e.g. chain.from_iterable([["pathone", "pathtwo"], [], ["paththree"]]) =>
+    # ("pathone", "pathtwo", "paththree")
+    sacred_dirs = itertools.chain.from_iterable(
+        sacred_util.filter_subdirs(source_dir) for source_dir in source_dirs
+    )
+    sacred_dicts = []
+
+    for sacred_dir in sacred_dirs:
+        try:
+            sacred_dicts.append(sacred_util.SacredDicts.load_from_dir(sacred_dir))
+        except json.JSONDecodeError:
+            warnings.warn(f"Invalid JSON file in {sacred_dir}", RuntimeWarning)
+
+    if run_name is not None:
```

Comment:
Consider splitting this into a helper function along the lines of:

```
def _filter_dicts(dicts, key, expected_val):
    return filter(lambda sd: get(sd.run, key) == expected_val, sacred_dicts)
```

Reply:
The helper pattern doesn't match `!= "FAILED"` on the `skip_failed_runs` conditional. (But otherwise I"d be happy with this change)